design.md

```diff
@@ -0,0 +1,73 @@
+# Bazel Module (bzlmod) Support in Renovate
+
+```yaml
+author: Chuck Grindel<chuck.grindel@gmail.com>
+status: To be reviewed
+created: 2023-04-10
+```
+
+## Abstract
+
+[Renovate] is a popular service for updating external dependencies. It [currently supports] updating
+Bazel dependencies described in [WORKSPACE-managed repositories]. With the introduction of [Bazel
+modules], [Renovate] needs to be modified to support the new external dependency mechanism.
+
+## Requirements
+
+- Support extracting external dependency information from `MODULE.bazel` files.
+  - Read external dependency information from [bazel_dep] declarations.
+  - Support module overrides.
+    - [Single version override]
+    - [Multiple version override]
+    - [Non-registry overrides]
+- Detect new module versions when they are introduced to a module registry (e.g., [Bazel Central
```

Comment:
Do you need to parse the project's `.bazelrc` to see if they are overriding the registry in use? Whilst we currently only have a single central registry it is concievable that in the future we end up with domain specific  'central' registries. Also for self-hosted internal registries and self hosted renovate we would want renovate to be able to work against those internal registries too.

Reply:
I thought that [the registry values](https://bazel.build/reference/command-line-reference#flag--registry) were going to be a configuration option. However, parsing the `.bazelrc` does seem like a better way to discover the value(s). The only complication I see is adding support for `imports` and `try-imports`. That is doable though. I will update the requirements to better reflect this. Thanks!